app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -87,26 +95,30 @@ def update_puppet_path
         path
       end
 
-      def pulp_node
-        @pulp_node ||= Katello::Pulp::Server.config(pulp_url, User.remote_user)
-      end
-
       def pulp_url
         uri = URI.parse(url)
         "#{uri.scheme}://#{uri.host}/pulp/api/v2/"
       end
 
-      def default_capsule?
+      def pulp_api
+        @pulp_api ||= Katello::Pulp::Server.config(pulp_url, User.remote_user)
+      end
+
+      def pulp_mirror?
+        self.features.map(&:name).include?(PULP_NODE_FEATURE)
+      end
+
+      def pulp_master?
```

Comment:
Can there be more than one pulp_masters? (renamed from default capsule?)

Pulp_master returns first record with feature Pulp whereas this checks only if feature pulp is present..Not sure if that would cause discrepancies..

Reply:
Currently there should not ever be more than one master.